builtin/providers/azurerm/encryption_settings.go

```diff
@@ -0,0 +1,163 @@
+package azurerm
+
+import (
+	"github.com/Azure/azure-sdk-for-go/arm/compute"
+	"github.com/Azure/azure-sdk-for-go/arm/disk"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func encryptionSettingsSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"enabled": {
+					Type:     schema.TypeBool,
+					Required: true,
+
+					// Azure can change enabled from false to true, but not the other way around, so
+					//   to keep idempotency, we'll conservatively set this to ForceNew=true
+					// TODO: Is this the right behavior?
+					ForceNew: true,
+				},
+
+				"disk_encryption_key": {
+					Type:     schema.TypeList,
```

Comment:
Okay, here's where things get unnecessarily complex.  I'm mimicking Azure's ARM schema here:

```hcl
  profile_os_disk {
    encryption_settings {
      enabled = true
      disk_encryption_key { ... }
      key_encryption_key { ... }
    }
  }
```
There's only ever one encryption_settings, but I can't see a good way to model this kind of nested object schema, except to use a set or list with MaxElem=1.  Using set seems like a really bad idea because of its weird semantics and bad diff'ing, so using a list seems the only available choice.

I could flatten this out to remove the nested objects, although that seems messy, too.

Advice?